apps/designer/app/shared/stores/styles.ts

```diff
@@ -0,0 +1,108 @@
+import { useEffect } from "react";
+import store from "immerhin";
+import { createValueContainer, useValue } from "react-nano-state";
+import type { StyleProperty, StyleValue } from "@webstudio-is/css-data";
+import type { Styles } from "@webstudio-is/react-sdk";
+import { publish, subscribe } from "~/shared/pubsub";
+import { useSyncInitializeOnce } from "~/shared/hook-utils";
+
+export const stylesContainer = createValueContainer<Styles>([]);
+
+export const useStyles = () => useValue(stylesContainer);
+
+type SetStylesMessage = {
+  store: "styles";
+  operation: "set";
+  breakpointId: string;
+  instanceId: string;
+  property: StyleProperty;
+  value: StyleValue;
+};
+
+type DeleteStylesMessage = {
+  store: "styles";
+  operation: "delete";
+  breakpointId: string;
+  instanceId: string;
+  property: StyleProperty;
+};
+
+export type StylesMessage = SetStylesMessage | DeleteStylesMessage;
```

Comment:
"StylesMesasge" as a name is confusing in different contexts, how about `StyleUpdateOperation`

Reply:
generally, "message" confuses me, in this case its a terminology from the underlying protocol, not high-level business level naming

Reply:
We already use operation as field for set and delete. Maybe just an StylesUpdate = SetStylesUpdate | DeleteStylesUpdate?